scripts/generate_dice_roll_html.rb

```diff
@@ -121,11 +121,14 @@
 mxes = [1, 1, -1, -1]
 myes = [1, -1, -1, 1]
 
+# https://phabricator.wikimedia.org/T350601
+rotation_offset = 0.0001 # Fixes a Safari CSS animation bug where the dice disappears
```

Comment:
N.B. This is the main fix - applying a small rotation offset here appears to fix the final animation keyframe. I then ran this Ruby code to re-generate the `WMFRandomDiceButton.html` file.

**Disclaimer: I am neither a Ruby nor a CSS developer (I am a Swift developer) so this is really just based on what I was able to find out. It seems these kinds of workarounds are commonplace with Safari anyway:**
* https://stackoverflow.com/questions/18146511/bug-in-css3-rotatey-transition-on-safari
* https://stackoverflow.com/questions/35186768/safari-css-bug-animation-rotation-direction-incorrect